### PR TITLE
feat(jetton-master): allow updating owner field

### PR DESCRIPTION
**Pull request by iamIcarus**
## PR Summary

- **Feature:** Added the ability to update the `owner` field in the Jetton Master contract using the `JettonSetParameter` message. Setting the owner to an empty address revokes ownership.
- **Tests:** Implemented comprehensive tests to verify:
  - Ownership updates.
  - Ownership revocation (setting to zero address).
- **Refactor:** 
  - Split the message definitions into separate files for better modularity.

### CODE REVIEW

File: contracts/jetton/master.tact [hunk 1]
```diff
@@ -88,4 +97,9 @@ contract JettonMaster with TEP74JettonMaster, TEP89JettonDiscoverable, Deployabl
     get fun get_tact_system_cell(): Cell {
         return self.jetton_wallet_system;
     }
+
+    //Returns the owner of the contract
+    get fun Owner(): Address {
```
Comment (iamIcarus) [inline contracts/jetton/master.tact:102]: this should be changed to lowercase. i will correct it

**Comment by ya7on** [comment]
Thank you! You made great PR!
Splitting messages is great idea, also let's update [README](https://github.com/supadupadao/jetton/tree/master?tab=readme-ov-file#project-structure) for new file structure

**Comment by iamIcarus** [comment]
> also let's update [README]

ah yes, ill make the change on that as well. I'll push the updates sometime tomorrow

**Comment by iamIcarus** [comment]
Not sure if we want to keep the ownership change through `JettonSetParameter`. The OwnableTransferable trait is defined in the [standard library](https://github.com/tact-lang/tact/blob/main/stdlib/libs/ownable.tact)

**Comment by ya7on** [comment]
Oh. I didn't know about that trait in Tact standard library. But it seems this message `ChangeOwner` not any kind of standard in TON, it's just non required helper function. It even doesn't have constant opcode.
So i believe we can use our own message for updating owner (using `JettonSetParameter`) because if we not, we will have overcomplicated API with one endpoint for some parameters and another endopoint for others.
What do you think about it?

**Comment by iamIcarus** [comment]
I found out about `OwnableTransferable` yesterday after implementing this PR 😄. 

I'm in favor of using `OwnableTransferable` because:

- If its in the standard lib, people are familiar with this and will probably want to use that.
- Already implements a reply with "ChangeOwnerOk"
- The `JettonSetParameter` might end up handling many different, unrelated functionalities with each having its own complex logic. I think its better if functionality that impacts the state of the contract (such us roles/ownership/mint control/lock/pause/tax ) would be on its own receive message:
  - We dont end up with a huge JettonSetParameter function/reducer
  - Scalability, not limited to a key/value
  - Better manage the project file structure with segregation of receive messages

**Comment by ya7on** [comment]
Ok then, I have to agree with you, let's use standard solution using `stdlib/ownable` for transfer ownership

**Comment by iamIcarus** [comment]
Alright, i'll implement the change and adjust the tests

**Comment by iamIcarus** [comment]
### PR Summary

#### Contract
- **Added** `OwnableTransferable` trait with `ChangeOwner` for ownership updates and revocations.

#### Tests
- **Test**  the use of `ChangeOwner` for ownership changes.
- **Test** ownership revocation via zero address.
- **Test** ownership through the minting functionality.

**Comment by ya7on** [comment]
Thank you!

**Pull request merged**